integration/README.md

```diff
@@ -4,36 +4,43 @@
+export CLUSTER_MANIFEST_FILE=$PWD/cluster.yaml
   ```
-    export AWS_REGION=us-east-1
-    export AWS_SSH_KEY_NAME=capi
-    export CONTROL_PLANE_MACHINE_COUNT=3
-    export WORKER_MACHINE_COUNT=3
-    export AWS_CREATE_BASTION=false
-    export AWS_PUBLIC_IP=false
-    export AWS_CONTROL_PLANE_MACHINE_FLAVOR=t3.large
-    export AWS_NODE_MACHINE_FLAVOR=t3.large
-    export CLUSTER_NAME=test-ci-cluster
-    clusterctl generate cluster ${CLUSTER_NAME} --from "templates/cluster-template-aws.yaml" --kubernetes-version 1.25.0 > cluster.yaml
-    export CLUSTER_MANIFEST_FILE=$PWD/cluster.yaml
-  ```
-
+> NOTE: AWS_SSH_KEY_NAME is name of ssh key in AWS that you are plan to use, if you don't have one yet refer
+> to capi on [aws prerequisites documentation](https://cluster-api-aws.sigs.k8s.io/topics/using-clusterawsadm-to-fulfill-prerequisites#ssh-key-pair)
   *  Additional environment variables when testing cluster upgrades:
-  ```
-    export CAPI_UPGRADE_VERSION=v1.26.0
-    export CAPI_UPGRADE_MD_NAME=${CLUSTER_NAME}-md-0
-    export CAPI_UPGRADE_MD_TYPE=machinedeployments.cluster.x-k8s.io
-    export CAPI_UPGRADE_CP_NAME=${CLUSTER_NAME}-control-plane
-    export CAPI_UPGRADE_CP_TYPE=microk8scontrolplanes.controlplane.cluster.x-k8s.io
-
-    # Change the control plane and worker machine count to desired values for in-place upgrades tests and create a new cluster manifest.
-    CONTROL_PLANE_MACHINE_COUNT=1
-    WORKER_MACHINE_COUNT=1
-    clusterctl generate cluster ${CLUSTER_NAME} --from "templates/cluster-template-aws.yaml" --kubernetes-version 1.25.0 > cluster-inplace.yaml
-    export CLUSTER_INPLACE_MANIFEST_FILE=$PWD/cluster-inplace.yaml
+  ```bash
+export CAPI_UPGRADE_VERSION=v1.28.0
+export CAPI_UPGRADE_MD_NAME=${CLUSTER_NAME}-md-0
+export CAPI_UPGRADE_MD_TYPE=machinedeployments.cluster.x-k8s.io
+export CAPI_UPGRADE_CP_NAME=${CLUSTER_NAME}-control-plane
+export CAPI_UPGRADE_CP_TYPE=microk8scontrolplanes.controlplane.cluster.x-k8s.io
+# Change the control plane and worker machine count to desired values for in-place upgrades tests and create a new cluster manifest.
+CONTROL_PLANE_MACHINE_COUNT=1
+WORKER_MACHINE_COUNT=1
+clusterctl generate cluster ${CLUSTER_NAME} --from "templates/cluster-template-aws.yaml" --kubernetes-version 1.27.0 > cluster-inplace.yaml
+export CLUSTER_INPLACE_MANIFEST_FILE=$PWD/cluster-inplace.yaml
+```
 
-  ```
+  *  Additional environment variables when testing disable default CNI flag:
+  ```bash
+export DISABLE_DEFAULT_CNI=true
+export POST_RUN_COMMANDS='["helm install cilium cilium/cilium  --namespace kube-system --set cni.confPath=/var/snap/microk8s/current/args/cni-network --set cni.binPath=/var/snap/microk8s/current/opt/cni/bin --set daemon.runPath=/var/snap/microk8s/current/var/run/cilium --set operator.replicas=1 --set ipam.operator.clusterPoolIPv4PodCIDRList=\"10.1.0.0/16\" --set nodePort.enabled=true"]' # install Calico in place default CNI
```

Comment:
why are you running this as a post run command?
I would just setup the cluster below and then do something like this:
```
	t.Run("DeployCluster", func(t *testing.T) { deployCluster(t, os.Getenv("CLUSTER_DISABLE_DEFAULT_CNI_MANIFEST_FILE")) })
	t.Run("ValidateCalico", func(t *testing.T) { validateCalico(t) })
	t.Run("DeployCilium", func(t *testing.T) { deployCilium(t) })
	t.Run("ValidateCilium", func(t *testing.T) { validateCilium(t) })
```

Reply:
I don't think the test should depend on the user setting env. variables. This will be confusing in the future when you want to run the whole test suite and the test suite depends on a bunch of variables and you have to go hunt for them. Instead, create a manifests directory, and hard code the file there, and load it from the test.